packages/@ember/-internals/deprecations/index.ts

```diff
@@ -0,0 +1,23 @@
+import type { DeprecationOptions } from '@ember/debug/lib/deprecate';
+import { ENV } from '@ember/-internals/environment';
+
+function isEnabled(options: DeprecationOptions) {
+  return Object.hasOwnProperty.call(options.since, 'enabled') || ENV._ALL_DEPRECATIONS_ENABLED;
+}
+
+function deprecation(options: DeprecationOptions) {
+  return {
+    options,
+    test: !isEnabled(options),
+    isEnabled: isEnabled(options),
+  };
+}
+export const DEPRECATIONS = {
```

Comment:
this might be tricky when it comes to me generating the list of things to deprecate for 'ember' -- maybe I'll codemod it.

Reply:
Can you elaborate?

Reply:
It should be the same list you needed to generate previously, but now will all be referenced from one location. The message can still vary by location used (and is expected to do so).